src/Dialog/Dialog.docs.json

```diff
@@ -37,14 +37,19 @@
     },
     {
       "name": "onClose",
-      "type": "(gesture: 'close-button' | 'escape') => void",
-      "description": "This method is invoked when a gesture to close the dialog is used (either an Escape key press or clicking the 'X' in the top-right corner). The gesture argument indicates the gesture that was used to close the dialog (either 'close-button' or 'escape')."
+      "type": "(gesture: 'close-button' | 'escape' | 'drag' | 'overlay') => void",
+      "description": "This method is invoked when a gesture to close the dialog is used (either an Escape key press, clicking/tapping on the backdrop, clicking/tapping the 'X' in the top-right corner or dragging away a bottom sheet). The gesture argument indicates the gesture that was used to close the dialog."
     },
     {
       "name": "role",
       "type": "'dialog' | 'alertdialog'",
       "description": "The ARIA role to assign to this dialog."
     },
+    {
+      "name": "type",
+      "type": "'default' | 'full-screen' | 'bottom-sheet'",
```

Comment:
I'm wondering if we should just make a `'bottom-sheet'` dialog it's own component instead of rendering it using a `Dialog`. Then, we could set `'full-screen'` in the `width` and `height` props. If `'full-screen'` is passed to _either_ `width` or `height`, the dialog would be full screen.

Curious to hear what other reviewers think.

Reply:
My concern is that you'll need to wrap different components in a `<Box/>` for responsive behaviour. For instance, we're planning to integrate side sheets shortly too. In this scenario, I'd prefer using a side sheet on desktop and a bottom sheet on mobile. This means I'll be required to proceed in this manner:

```javascript
<Box sx={{display: ['block', 'block', 'none']}}><BottomSheet /></Box>
<Box sx={{display: ['none', 'none', 'block']}}><SideSheet /></Box>
``` 

Another situation where I need to follow this approach is when I want to display a bottom sheet on mobile, but a standard dialog on desktop.

```javascript
<Box sx={{display: ['block', 'block', 'none']}}><BottomSheet /></Box>
<Box sx={{display: ['none', 'none', 'block']}}><Dialog /></Box>
``` 

This adds extra complexity because an open `<Dialog/>` that turns into `<BottomSheet/>` on mobile would need to stay open. If the user applies separate states for each component, it might lead to unintended effects.

The proposed API really simplifies above and ensures consumers still understand the underlying semantics is a dialog.

```javascript
<Dialog title="Your title" type={{
    narrow: 'full-screen',
    regular: 'default',
    wide: 'default'
}} >
```

or

```javascript
<Dialog title="Your title" type={{
    narrow: 'bottom-sheet',
    regular: 'default',
    wide: 'default'
}} >
```